corehq/apps/reports/util.py

```diff
@@ -136,9 +136,8 @@ def _create_temp_user(user_id):
 
         if user_filter[HQUserType.REGISTERED].show:
             # now add all the registered users who never submitted anything
-            for user_id in registered_user_ids:
+            for user_id, user in registered_user_ids.items():
```

Comment:
I'd go with `registered_users_by_id` to indicate it's a dict.  @millerdev and I have a date to refactor this whole function, it's one of those things that started small, then grew and grew until it does like 3 or 4 unrelated things, is really unperformant, and is hard to reason about.